### PR TITLE
Refactored show tool window command handlers to a single generic command

### PR DESCRIPTION
**Pull request by duncanp-sonar**

### CODE REVIEW

File: src/IssueViz.Security/Commands/ShowToolWindowCommand.cs [hunk 1]
```diff
@@ -20,48 +20,48 @@
...
 using Microsoft.VisualStudio.Shell;
 using SonarLint.VisualStudio.Core;
 using SonarLint.VisualStudio.Integration;
-using SonarLint.VisualStudio.IssueVisualization.Security.UI.HotspotsList;
-using Task = System.Threading.Tasks.Task;
 
 namespace SonarLint.VisualStudio.IssueVisualization.Security.Commands
 {
-    internal sealed class HotspotsToolWindowCommand
+    internal sealed class ShowToolWindowCommand
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/Commands/ShowToolWindowCommand.cs:31]: The GitHub is showing the new class as a modification to one of the existing ones. Still, perhaps that makes it easier to see the differences between them.

File: src/IssueViz.Security/IssueVizSecurityPackage.cs [hunk 2]
```diff
@@ -47,8 +48,16 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
         {
             await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            await HotspotsToolWindowCommand.InitializeAsync(this);
-            await TaintToolWindowCommand.InitializeAsync(this);
+            // We're not storing references to the command handler instances.
+            // We relying on the fact that the command handler instance registers a
+            // callback with the menu service to stop it from being garbage collected.
```
Comment (duncanp-sonar) [inline src/IssueViz.Security/IssueVizSecurityPackage.cs:53]: I'm slightly worried about this, but logically it makes senses; the instances are reachable so they're not eligible for collection.

File: src/IssueViz.Security/Commands/ShowToolWindowCommand.cs [hunk 3]
```diff
@@ -70,11 +70,11 @@ public static async Task InitializeAsync(AsyncPackage package)
 
             try
             {
-                toolWindowService.Show(HotspotsToolWindow.ToolWindowId);
+                toolWindowService.Show(toolWindowId);
             }
             catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
             {
-                logger.WriteLine(string.Format(Resources.ERR_HotspotsToolWindow_Exception, ex));
+                logger.WriteLine(string.Format(Resources.ERR_ShowToolWindow_Exception, ex));
```
Comment (rita-g-sonarsource) [inline src/IssueViz.Security/Commands/ShowToolWindowCommand.cs:77]: I think it's worth printing the toolWindowId or the commandId, anything that will help us identify which window was it that failed.

**Review by rita-g-sonarsource — APPROVED** [review]
Great LGTM. Only one comment regarding the log message.

**Pull request merged**